code/main/media/mediaModule.brs

```diff
@@ -0,0 +1,245 @@
+
+                    ' if the session id is not empty, update the session id and process the queued requests
+                    ' the location hint is optional, if it is empty, it will be ignored
+                    if not _adb_isEmptyOrInvalidString(sessionId)
+                        queuedMediaRequests = m._sessionManager.updateSessionIdAndGetQueuedRequests(requestId, sessionId, location)
+                        for each mediaRequest in queuedMediaRequests
+                            mediaEventType = mediaRequest.xdmData.xdm.eventType
+                            m._handleMediaEvent(mediaEventType, mediaRequest.requestId, sessionId, location, mediaRequest.xdmData, mediaRequest.tsObject)
+                        end for
+                        m._kickRequestQueue()
+                    end if
+                end if
+            end for
+        end sub,
+
+        _handleMediaEvent: sub(mediaEventType as string, requestId as string, sessionId as string, location as string, xdmData as object, tsObject as object)
+            path = _adb_EdgePathForEventType(mediaEventType, location, m._CONSTANTS.MEDIA.EVENT_TYPES)
+            if not _adb_isEmptyOrInvalidString(path)
+                xdmData.xdm["_id"] = _adb_generate_UUID()
+                xdmData.xdm["timestamp"] = tsObject.ts_inISO8601
+                xdmData.xdm["mediaCollection"]["sessionID"] = sessionId
+
+                meta = {}
+                ' TODO: sanitize the xdmData object before sending to the backend.
+                m._edgeRequestWorker.queue(requestId, [xdmData], tsObject.ts_inMillis, meta, path)
+                m._kickRequestQueue()
+            else
+                _adb_logError("_handleMediaEvent() - mediaEventName is invalid: " + mediaEventType)
+            end if
+        end sub,
+
+        _getEdgeConfig: function() as object
+            configId = m._configurationModule.getConfigId()
+            if _adb_isEmptyOrInvalidString(configId)
+                return invalid
+            end if
+            ecid = m._identityModule.getECID()
+            if _adb_isEmptyOrInvalidString(ecid)
+                return invalid
+            end if
+            return {
+                configId: configId,
+                ecid: ecid,
+                edgeDomain: m._configurationModule.getEdgeDomain()
+            }
+        end function,
+
+        _processQueuedRequests: function() as dynamic
+            responseEvents = []
+
+            if not m._edgeRequestWorker.hasQueuedEvent()
+                ' no requests to process
+                return responseEvents
+            end if
+
+            edgeConfig = m._getEdgeConfig()
+            if edgeConfig = invalid
+                _adb_logVerbose("_processQueuedRequests() - Cannot send network request, invalid configuration.")
+                return responseEvents
+            end if
+
+            responses = m._edgeRequestWorker.processRequests(edgeConfig.configId, edgeConfig.ecid, edgeConfig.edgeDomain)
```

Comment:
New task:  we should refactor our edge code so that other extension dont need to access edge configuration in order to send the edge requests, that should be handled by edge. All the hits can be queued in the edge module and sent when edge has valid configuration. It doesnt look right to access edge configuration and ecid here.

Reply:
Yeah, I was thinking the same. Let me add a TODO here.